src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -98,30 +90,39 @@ class AppsResource @Inject() (
   @Timed
   def show(@PathParam("id") id: String,
            @QueryParam("embed") embed: java.util.Set[String],
-           @Context req: HttpServletRequest,
-           @Context resp: HttpServletResponse): Response = {
-    doIfAuthorized(req, resp, ViewAppOrGroup, id.toRootPath) { identity =>
-      val resolvedEmbed = AppInfoEmbedResolver.resolve(embed) ++ Set(
-        // deprecated. For compatibility.
-        AppInfo.Embed.Counts, AppInfo.Embed.Tasks, AppInfo.Embed.LastTaskFailure, AppInfo.Embed.Deployments
-      )
-      def transitiveApps(gid: PathId): Response = {
-        val withTasks = result(appInfoService.queryAllInGroup(gid, resolvedEmbed))
-        ok(jsonObjString("*" -> withTasks))
-      }
-      def app(): Future[Response] = {
-        val maybeAppInfo = appInfoService.queryForAppId(id.toRootPath, resolvedEmbed)
+           @Context req: HttpServletRequest): Response = doIfAuthenticated(req) { implicit identity =>
+    val resolvedEmbed = AppInfoEmbedResolver.resolve(embed) ++ Set(
+      // deprecated. For compatibility.
+      AppInfo.Embed.Counts, AppInfo.Embed.Tasks, AppInfo.Embed.LastTaskFailure, AppInfo.Embed.Deployments
+    )
 
-        maybeAppInfo.map {
-          case Some(appInfo) => ok(jsonObjString("app" -> appInfo))
-          case None          => unknownApp(id.toRootPath)
-        }
+    def transitiveApps(groupId: PathId): Response = {
+      result(groupManager.group(groupId)) match {
+        case Some(group) =>
+          doIfAuthorized(ViewGroup, group) {
```

Comment:
The problem with my change is that it will return all transitive apps in a group, but authorizes only the view on the first group, not on the subgroups as well.